repos/system_upgrade/el7toel8/actors/kernel/checkkerneldrivers/actor.py

```diff
@@ -16,59 +16,51 @@ class CheckKernelDrivers(Actor):
     Note:
      - List of kernel drivers missing on the RHEL8 system is located in the
         files/removed_drivers.txt file.
-     - Whitelisted modules that are not going to be reported are consumed from
-        the WhitelistedKernelModules
     """
     name = 'check_kernel_drivers'
-    consumes = (ActiveKernelModulesFacts, WhitelistedKernelModules)
+    consumes = (PCIDevices,)
     produces = (Report,)
     tags = (ChecksPhaseTag, IPUWorkflowTag)
 
     def process(self):
-        with open('files/removed_drivers.txt', 'r') as removed:
-            removed_drivers = []
-            whitelisted_modules = set()
-            collected_drivers = set()
+        removed_drivers = set()
+        present_drivers = set()
+        conflicting_drivers = set()
 
+        with open('files/removed_drivers.txt', 'r') as removed:
```

Comment:
exception handling required, to harden the code

Plus, since you rewrite this actor I'd recommend to move this to a library so you can better unit test the code.

Reply:
Alright, handled exceptions and refactored the code a bit.